### ISSUE TITLE
[BUG] $pull + $in remove all embedded documents in an array type field.

### ISSUE THREAD

**Issue by FrostyLeaf** [Automattic/mongoose#6439]
**Node Version**: 8.9.4

**Mongoose Version**: 5.0.17; 4.13.12

**Mongodb Version**: 3.2.9

This is how to reproduce this bug:

```js
const mongoose = require('mongoose');

mongoose.connect('mongodb://localhost/test', {debug: true});
mongoose.set('debug', true);

const Cat = mongoose.model('Cat', {
  name: String,
  props: [
    new mongoose.Schema(
      {
        name: String
      },
      {_id: false}
    )
  ]
});

(async function () {
  let kitty = new Cat({name: 'Zildjian',
    props: [
      {name: 'invalid'},
      {name: 'abc'},
      {name: 'def'}
    ]
  });
  await kitty.save();

  kitty = await Cat.findById(kitty._id);
  console.log(kitty.props);

  // The setter is casted as `{ '$pull': { props: {} } }`, which leads the all embedded documents of `props` field to be removed.
  await Cat.update({_id: kitty._id}, {$pull: {props: {$in: [{name: 'invalid'}, {name: 'def'}]}}});

  kitty = await Cat.findById(kitty._id);
  console.log('result:', kitty.props); // Expect an array with 1 embedded document inside, but get 0 here.
}());

```

**Expected Behavior**: `$pull` + `$in` should only remove the embedded documents match the assigned conditions, not all of them.

**Comment by FrostyLeaf** [comment]
Any one help? @vkarpov15

**Comment by lineus** [comment]
@FrostyLeaf thanks for the concise report!

**Comment by FrostyLeaf** [comment]
So, when v5.0.18 will be published, please? And will you guys fix this on v4?

**Comment by vkarpov15** [comment]
Monday is when we are planning on releasing this for 5.x. We will backport this to 4.x as well

**Comment by lineus** [comment]
My bad, I meant to reopen this when I saw the request for v4, I must have archived the email unintentionally. PR submitted.

**Comment by vkarpov15** [comment]
Fixed by #6467, will release 4.13.13 tomorrow
